go/vt/wrangler/reparent.go

```diff
@@ -763,9 +797,11 @@ func (maxPosSearch *maxReplPosSearch) processTablet(tablet *topodatapb.Tablet) {
 		maxPosSearch.wrangler.logger.Warningf("failed to get replication status from %v, ignoring tablet: %v", topoproto.TabletAliasString(tablet.Alias), err)
 		return
 	}
-	replPos, err := mysql.DecodePosition(status.Position)
+	// The replica that is most progressed may actually be the one with the furthest
```

Comment:
I think we should check both positions and use the one that's a superset of the other. If neither is a true superset, then we have detected errant transactions and it isn't safe to promote this replica.

Reply:
Deferred to a future PR.